x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/planner/ExpressionTranslators.java

```diff
@@ -336,15 +337,15 @@ static Query translate(BinaryComparison bc, TranslatorHandler handler) {
             if (bc instanceof LessThanOrEqual) {
                 return new RangeQuery(source, name, null, false, value, true, format, zoneId);
             }
-            if (bc instanceof Equals || bc instanceof NullEquals || bc instanceof NotEquals) {
+            if (bc instanceof Equals || bc instanceof NullEquals || bc instanceof NotEquals || bc instanceof EqualsIgnoreCase) {
```

Comment:
No need to touch QL - see InsensitiveBinaryComparisons inside EQL on how to extend the method (add a dedicated TranslatorHandler).